src/components/KPI/kpiProfit.jsx

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+export default function KpiProfit({ qlikApp, objectId, title, color }) {
+  // Define states
+  const [kpiObject, setKpiObject] = useState();
+
+  // Get the kpiObject, using the props provided and set to state
+  useEffect(() => {
+    // Get the kpiObject, which is an instance of a qlik object & set to state
+    const getSetObjectDetails = async () => {
+      // Get the specified object from the instance of the specific app
+      const qObject = await qlikApp.getObject(objectId);
+      setKpiObject(qObject);
+    };
+    // Only call the getSetObject when: kpiObject doesn't exist, qlikApp does exist and objectId does exist
+    if (!kpiObject && qlikApp && objectId) {
+      getSetObjectDetails();
+    }
+  }, [kpiObject, qlikApp, objectId]);
+
+  // Get the relevant kpiData from the layout of the kpiObject and set to state
+  const [kpiData, setKpiData] = useState(undefined);
+  useEffect(() => {
+    const getKpiData = async () => {
+      const qPath = '/qHyperCubeDef';
+      const qPages = [
+        {
+          qLeft: 0,
+          qTop: 0,
+          qWidth: 4, // layoutObject['qHyperCube']['qSize']['qcx']
```

Comment:
FYI (no change necessary):
you could do a getLayout call to dynamically source the qWidth and qHeight (which is the qcx & qcy in the layout object) so if the qlik developer ever changed the number of rows, your code could handle it without needing an update.